### PR TITLE
Cache component query results

### PR DESCRIPTION
**Pull request by mawildoer**
# Picker: Cache ComponentQuery

# Description
Cache queries because in a normal design a lot of same components are picked.

Fixes # (issue)

# Checklist

Please read and execute the following:

- [ ] My code follows the [coding guidelines](/faebryk/faebryk/blob/main/docs/CODING_GUIDELINES.md) of this project
- [ ] My PR title is following the [contribution guidelines](/faebryk/faebryk/blob/main/docs/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I ran [Black](../docs/CONTRIBUTING.md#creating-a-pull-request) to format my code

#### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](/faebryk/faebryk/blob/main/docs/CODE_OF_CONDUCT.md):

- [ ] I agree to follow this project's [Code of Conduct](/faebryk/faebryk/blob/main/docs/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by IoannisP-ITENG** [comment]
Have you ran it with `./tools/profile.sh py pyinstrument <bikey-wagon>'?
Looks like most time is spent in the python part (not sql part). We can cache that too.

**Comment by mawildoer** [comment]
Thanks for the link! I’ll give it another swing with the profiler when I
get the chance

File: src/faebryk/libs/picker/jlcpcb/jlcpcb.py [hunk 1]
```diff
@@ -378,7 +381,14 @@ async def exec(self) -> list[Component]:
         return self.results
 
     def get(self) -> list[Component]:
-        return self.results or asyncio.run(self.exec())
+        if self.results:
+            return self.results
+
+        query_hash = str(self.Q)
```
Comment (iopapamanoglou) [inline src/faebryk/libs/picker/jlcpcb/jlcpcb.py:387]: This does not work:
`<tortoise.expressions.Q object at 0x742078c0d4f0>`

**Review by iopapamanoglou — CHANGES_REQUESTED** [review]
This won't work like this, we need to do per filter caching.

**Comment by iopapamanoglou** [comment]
Superseded by CachePicker

**Pull request closed without merging**